### PR TITLE
App: Add `xr_hello_holoscan` demo application

### PR DESCRIPTION
**Pull request by tbirdso**
Adds a lightweight demo application to showcase Holoscan SDK + Magic Leap operations. The demo application presents a basic scene with primitive geometries and live controller or hand tracking sensor outputs to showcase available inputs.

The application relies on the container infrastructure previously established in the `volume_rendering_xr` demo application.

This work is migrated from an internal development collaboration between the Magic Leap and NVIDIA Holoscan teams. Contributors are credited in the application metadata file.

### CODE REVIEW

**Review by jjomier — APPROVED** [review]
No way to test it, but it looks good.

**Comment by tbirdso** [comment]
@jjomier there is a debugger UI you can use to test without a device. Let me update the README to reference the instructions in `volume_rendering_xr`.

**Review by AndreasHeumann — APPROVED** [review]
One nit-pick (sorry). Else looks good!

**Comment by AndreasHeumann** [comment]
I also tested the application locally and it works nicely.

File: applications/xr_hello_holoscan/metadata.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+					"url": "https://www.magicleap.com/magic-leap-2"
+				}
+			],
+			"libraries": [
+				{
+				"name": "windrunner",
+				"description": "Magic Leap OpenXR native backend",
+				"version": "1.11.73",
+				"license": "Magic Leap 2 Software Agreement",
+				"license-url": "https://www.magicleap.com/software-license-agreement-ml2"
```
[the remark this answers is not in the document]
Comment (tbirdso) [reply applications/xr_hello_holoscan/metadata.json]: Fixed!

**Pull request merged**